### PR TITLE
Deprecate ScatterVisualizer to contrib

### LINKED ISSUE #440: deprecate ScatterPlotVisualizer
- ``ScatterPlotVisualizer`` is being moved to contrib in 0.8
  - ``DecisionBoundaryVisualizer`` is being moved to contrib in 0.8

### PR DESCRIPTION
**Pull request by bbengfort**
To maintain backward compatibility, the ScatterVisualizer has been moved
to contrib, so that the new ScatterPlotMixin (which borrows heavily from
this visualizer) does not break any user code.

Moved the ScatterVisualizer to contrib, updated documentation and
tests. No changes were made to the visualizer, but it can be updated to
use the new ScatterPlotMixin.

Note that documentation now lists "direct data visualization" as only
the JointPlotVisualizer, but we should add some of the single feature
visualizations to direct data visualization as well.

Fixes #440

### CODE REVIEW

**Comment by bbengfort** [comment]
@ndanielsen FYI - this is in preparation for the scatter visualizer mixin.

**Comment by lwgray** [comment]
What's up with the failed image?

**Comment by bbengfort** [comment]
`test_scatter_image_fail.png` is one of the tests that asserts when the ScatterVisualizer errors, it does so gracefully but doesn't produce an image.

**Comment by zjpoh** [comment]
examples.ipynb imports ScatterVisualizer from `yellowbrick.features.scatter`. Should we update that to import from `yellowbrick.contrib.scatter` also?

**Comment by ndanielsen** [comment]
@zjpoh that's a great suggestion!

File: yellowbrick/features/__init__.py [hunk 1]
```diff
@@ -21,7 +21,6 @@
 from .pcoords import ParallelCoordinates, parallel_coordinates
 from .radviz import RadialVisualizer, RadViz, radviz
 from .rankd import Rank1D, rank1d, Rank2D, rank2d
-from .scatter import ScatterViz, ScatterVisualizer, scatterviz
```
Comment (rebeccabilbro) [inline yellowbrick/features/__init__.py]: Hey @bbengfort - it looks like we removed this import line from `yellowbrick/features/__init__.py` but maybe didn't add it to `yellowbrick/contrib/__init__.py`?

Comment (bbengfort) [reply yellowbrick/features/__init__.py]: Hi @rebeccabilbro sorry, I didn't see this until just now. I guess I didn't really intend to hoist the visualizers into the `contrib` package; was envisioning that each `contrib` package would be isolated (e.g. to protect optional dependencies). However, in this case, if you feel like our users would benefit from being able to more directly import the `ScatterVisualizer`, please feel free to add that import statement.

Comment (rebeccabilbro) [reply yellowbrick/features/__init__.py]: Ok, will do!

**Pull request merged**